File: src/AbstractFactory/Models/Car.java
package AbstractFactory.Models;

import AbstractFactory.Factories.IFactory;
import AbstractFactory.Products.Bonnet.IBonnet;
import AbstractFactory.Products.Chassis.IChassis;
import AbstractFactory.Products.Door.IDoor;
import AbstractFactory.Products.Motor.IMotor;
import AbstractFactory.Products.Seat.ISeat;
import AbstractFactory.Products.SteeringWheel.ISteeringWheel;
import AbstractFactory.Products.TailPipe.ITailPipe;
import AbstractFactory.Products.Wheel.IWheel;

public class Car implements IVehicle{
    private IWheel[] wheels = new IWheel[4];
    private IDoor[] doors = new IDoor[4];
    private IBonnet bonnet;
    private IChassis chassis;
    private IMotor motor;
    private ISeat[] seats = new ISeat[3];
    private ISteeringWheel steeringWheel;
    private ITailPipe tailPipe;

    public Car(IFactory factory){
        this.wheels[0] = factory.createWheel(17, 3);
        this.wheels[1] = factory.createWheel(17, 3);
        this.wheels[2] = factory.createWheel(17, 3);
        this.wheels[3] = factory.createWheel(17, 3);
        this.doors[0] = factory.createDoor();
        this.doors[1] = factory.createDoor();
        this.doors[2] = factory.createDoor();
        this.doors[3] = factory.createDoor();
        this.bonnet = factory.createBonnet(160);
        this.chassis = factory.createChassis();
        this.motor = factory.createMotor();
        this.seats[0] = factory.createSeat("negro", "70x90");
        this.seats[1] = factory.createSeat("negro", "70x90");
        this.seats[2] = factory.createSeat("negro", "180x90");
        this.steeringWheel = factory.createSteeringWheel();
        this.tailPipe = factory.createTailPipe();
    }

    public String getVehicleInformation(){
        return String.format("" +
            "El auto cuenta con %d ruedas caracterizadas por: %s. %n" +
            "Es un auto de %d puertas caracterizadas por: %s. %n" +
            "Contiene %d asientos. %s. %n" +
            "%s. %n" +
            "%s. %n" +
            "%s. %n" +
            "%s. %n" +
            "%s.%n",
            this.wheels.length, this.wheels[0].getInformation(),
            this.doors.length, this.doors[0].getInformation(),
            this.seats.length, this.seats[0].getInformation(),
            this.bonnet.getInformation(),
            this.chassis.getInformation(),
            this.motor.getInformation(),
            this.steeringWheel.getInformation(),
            this.tailPipe.getInformation());
    }
}

File: src/Builder/Models/Car.cs
using Builder.Products;
using System.Collections.Generic;
using System;

namespace Builder.Models{

public class Car : IVehicle{
    public List<IProduct> wheels {get; set;}
    public List<IProduct> doors {get; set;}
    public IProduct bonnet {get; set;}
    public IProduct chassis {get; set;}
    public IProduct motor {get; set;}
    public List<IProduct> seats {get; set;}
    public IProduct steeringWheel {get; set;}
    public IProduct tailPipe {get; set;}
    public IProduct frame {get; set;}
    public IProduct handlebars {get; set;}
    public IProduct chain {get; set;}

    public Car(){
        this.wheels = new List<IProduct>();
        this.doors = new List<IProduct>();
        this.seats = new List<IProduct>();
    }

    public String getVehicleInformation(){
        return String.Format("" +
            "El auto cuenta con {0} ruedas caracterizadas por: {1}. " + Environment.NewLine + "" +
            "Es un auto de {2} puertas caracterizadas por: {3}. " + Environment.NewLine + "" +
            "Contiene {4} asientos. {5}. " + Environment.NewLine + "" +
            "{6}. " + Environment.NewLine + "" +
            "{7}. " + Environment.NewLine + "" +
            "{8}. " + Environment.NewLine + "" +
            "{9}. " + Environment.NewLine + "" +
            "{10}." + Environment.NewLine + "",
            this.wheels.Count, this.wheels[0].getInformation(),
            this.doors.Count, this.doors[0].getInformation(),
            this.seats.Count, this.seats[0].getInformation(),
            this.bonnet.getInformation(),
            this.chassis.getInformation(),
            this.motor.getInformation(),
            this.steeringWheel.getInformation(),
            this.tailPipe.getInformation());
    }
}
}